### PR TITLE
Feature/kas 3874 search highlighting

### PR DESCRIPTION
**Pull request by kobemertens**
:warning: Needs https://github.com/kanselarij-vlaanderen/app-kaleidos/pull/366 for the backend

### CODE REVIEW

**Comment by sergiofenoll** [comment]
mu-search v0.9.0-beta.2 currently has a bug. It breaks the search index (file updates won't get handled) and our tests correctly caught that! A PR is open to fix that, beta.3 should be released soon so we can get this merged.

**Comment by ValenberghsSven** [comment]
the base of this branch is feature/KAS-3796-result-cards but there is no PR related for that branch that merges into development.

File: app/styles/auk/_auk-additions.scss [hunk 1]
```diff
@@ -339,3 +339,10 @@ table tr.lt-expanded-row {
 .auk-u-text-vertical-align-middle {
   vertical-align: middle;
 }
+
+// Result cards
+// Workaround until we adapt mu-search to send more specific tags for highlights
+.result-card em {
+  background: rgba($auk-yellow-500, .3);
```
Comment (tdn) [inline app/styles/auk/_auk-additions.scss]: Following Hannes' reply here: https://chat.semte.ch/group/kanselarij-dev?msg=dHssARyX36JnyXRkB I would remove this line. If user feedback shows that the bold font is not clear enough, they can decide on a color that's not used for other purposes

File: app/components/agenda/agendaitem/agendaitem-decision.js [hunk 2]
```diff
@@ -77,7 +77,7 @@ export default class AgendaitemDecisionComponent extends Component {
     const defaultAccessLevel = await this.store.findRecordByUri(
       'concept', subcaseIsConfidential
         ? CONSTANTS.ACCESS_LEVELS.VERTROUWELIJK
-        : CONSTANTS.ACCESS_LEVELS.INTERN_REGERING
+        : CONSTANTS.ACCESS_LEVELS.INTERN_OVERHEID
```
Comment (tdn) [inline app/components/agenda/agendaitem/agendaitem-decision.js]: I assume this is from a different PR that hasn't been merged yet?

File: app/templates/search/documents.hbs [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        @message={{t "fill-in-search-term"}}
+      />
+    {{else}}
+      <Auk::EmptyState
+        @bordered={{true}}
+        @message={{t "no-results-found-modify-term"}}
+      />
+    {{/if}}
+  {{/if}}
+</div>
```
Comment (tdn) [inline app/templates/search/documents.hbs]: Not sure if our linter picks up on missing newline at EOF, but perhaps add it for completeness sake...

**Review by tdn — CHANGES_REQUESTED** [review]
Some small things, and the document highlight is still in progress so waiting for that before merging. 

Decision and newsitem highlighting did not work on this branch, but did in the follow-up branch for https://github.com/kanselarij-vlaanderen/frontend-kaleidos/pull/1653 , so is probably a local version issue on my end.

The "search-in-decisions-only" filter may be removed in the cases search template: https://github.com/kanselarij-vlaanderen/frontend-kaleidos/pull/1633/files#diff-5e211f40b3c8f4b27f1b17d3f8ca1b93c23944ccc4df95781423eb1846236891R3-R17

Everything else looks good in local testing, a major improvement on the old search!

Code looks clear and readable, saw no weird things, thanks for that :-)

Make sure https://github.com/kanselarij-vlaanderen/frontend-kaleidos/pull/1653 is merged into this branch before merging this PR.

File: app/components/agenda/agendaitem/agendaitem-decision.js (hunk 2 above)
Comment (ValenberghsSven) [reply app/components/agenda/agendaitem/agendaitem-decision.js]: default of a decision is `INTERN_OVERHEID` from https://kanselarij.atlassian.net/browse/KAS-3826. Is on DEV.
I think this is a problem that the base branch is not up to date with development, but this branch is.
Goes back to my comment "the base of this branch is feature/KAS-3796-result-cards but there is no PR related for that branch that merges into development."

Correct flow should be = update base branch with development > update forked branch with base branch.
It seems we are doing = update forked branch with development which causes this kind of diffs

Comment (ValenberghsSven) [reply app/components/agenda/agendaitem/agendaitem-decision.js]: Going to do this now, this PR diff is bothering me

File: app/styles/auk/_auk-additions.scss (hunk 1 above)
Comment (kobemertens) [reply app/styles/auk/_auk-additions.scss]: I removed this in [29390bc](https://github.com/kanselarij-vlaanderen/frontend-kaleidos/pull/1633/commits/29390bc56c183fc159567d2a9baabed8e518cf9c)

File: app/components/search/result-cards/document.hbs [hunk 4]
```diff
@@ -1,27 +1,38 @@
...
       <AuLink
         class="au-u-padding-none"
-        @route="agenda.agendaitems.agendaitem"
+        @route="document"
         @model={{@documentId}}
+        @icon="document"
+        @iconAlignment="left"
       >
-        {{@title}}
+        {{@fileName}}
```
Comment (sergiofenoll) [inline app/components/search/result-cards/document.hbs:16]: Document card shows the file name, not the piece title. Is this intentional?

image

Comment (tdn) [reply app/components/search/result-cards/document.hbs:16]: Let's take this up with Hannes at 16:00.
He was going to get back to us on what goes where in the document card, but unless I missed it, he never did, so I told Kobe to leave it as is for now. Small change if needed

File: translations/nl-be.json [hunk 5]
```diff
@@ -1155,5 +1155,9 @@
   "removed-cases": "Verwijderde dossiers",
   "open": "Open",
   "out": "Uit",
-  "found-inside": "Gevonden in"
+  "found-inside": "Gevonden in",
+  "found-in-thing": "Gevonden in {thing}",
+  "and-one-other-subcase": "en nog 1 andere procedurestap",
+  "and-n-other-subcases": "en nog {n} andere procedurestappen",
+  "uploaded-in-thing": "Geupload in {thing}"
```
Comment (sergiofenoll) [inline translations/nl-be.json:1163]: Shouldn't this be "Geüpload"? I think that's the correct spelling.

File: app/components/search/result-cards/document.hbs (hunk 4 above)
Comment (ValenberghsSven) [reply app/components/search/result-cards/document.hbs:16]: I would go for piece name as that is editable, file name is not.

Comment (tdn) [inline app/components/search/result-cards/document.hbs:16]: Pretty sure this needs a SanitizeHtml around it. Will fix on dev with any other bugs so we can merge now

**Review by ValenberghsSven — APPROVED** [review]
it's merging time.
Any small fixes may be done on DEV.

**Pull request merged**